src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -326,20 +349,75 @@ class RemoteMachineTrainingService implements TrainingService {
 
         return deferred.promise;
     }
-
-    public cleanUp(): Promise<void> {
+    
+    /**
+     * cleanup() has a time out of 10s to clean remote connections 
+     */
+    public async cleanUp(): Promise<void> {
         this.log.info('Stopping remote machine training service...');
         this.stopping = true;
+        let timeoutId: NodeJS.Timer;
+        const delay: Promise<{}> = new Promise((resolve: Function, reject: Function): void => {
+            timeoutId = setTimeout(
+                () => { reject(new Error('TrainingService cleanup timeout.')); },
+                10000);
+        });
+        await Promise.race([delay, this.cleanupConnections()]).finally(() => {
+            clearTimeout(timeoutId);
+        });
+    }
+    
+    /**
+     * stop gpu_metric_collector process in remote machine and remove unused scripts
+     */
+    private async cleanupConnections(): Promise<void> {
+        try{
+            for (const [rmMeta, client] of this.machineSSHClientMap.entries()) {
+                let jobpidPath: string = path.join(this.getRemoteScriptsPath(), 'pid');
+                await SSHClientUtility.remoteExeCommand(`pkill -P \`cat ${jobpidPath}\``, client);
+                await SSHClientUtility.remoteExeCommand(`rm -rf ${this.getRemoteScriptsPath()}`, client);
+            }
+        }catch (error) {
+            this.log.error(`Cleanup connection exception, error is ${error.message}`);
```

Comment:
Normally you should not catch Error and do not throw it in training services,  if you do this,  add comments to explain why.

Reply:
add comments.